### ISSUE TITLE
Automated renewal

### ISSUE THREAD

**Issue by somewhatnmc** [namecoin/namecoin-core#319]
Missing a function of automated name updates under a specific threshold (e.g. at 20 blocks remaining). Suggested in #305 by @Tradefreak, continued here.

**Comment by domob1812** [comment]
I agree that would be useful.  However, I'm not sure if that should be part of the core daemon or perhaps rather something done by the Qt "frontend".  @JeremyRand, what do you think?

Also this is a bit complicated with encrypted wallets - for those we would need to pre-create and store (likely in the wallet) the update transaction, but then also make sure that it doesn't get invalidated by a double spend in the mean time.  This could be solved by always pre-creating transactions for all names in the wallet whenever a transaction is done (and the wallet unlocked), but that seems both a potential performance issue and messy.

**Comment by yanmaani** [comment]
Transaction queuing has been considered, and is already added to electrum-nmc. That functionality is/shall be used for `name_autorenew` and `name_autoregister` in that project. Those are to be added in Core too, as I understand it.
